stock_whole_kit_constraint/models/stock_move.py

```diff
@@ -0,0 +1,68 @@
+# Copyright 2021 Tecnativa - David Vidal
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from odoo import api, fields, models
+
+
+class StockMove(models.Model):
+    _inherit = "stock.move"
+
+    allow_partial_kit_delivery = fields.Boolean(
+        compute="_compute_allow_partial_kit_delivery",
+        compute_sudo=True,
+        default=True,
+    )
+
+    @api.depends(
+        "product_id.product_tmpl_id.allow_partial_kit_delivery", "state"
+    )
+    def _compute_allow_partial_kit_delivery(self):
+        """Take it from the product only if it's a kit"""
+        for move in self.filtered(
+            lambda x: x.product_id and x.state not in ["done", "cancel"]
+        ):
+            # If it isn't a kit it will always be True
+            if (
+                not move.bom_line_id or move.bom_line_id.bom_id.type != "phantom"
+            ):
+                move.allow_partial_kit_delivery = True
+                continue
+            move.allow_partial_kit_delivery = (
+                move.bom_line_id.bom_id.product_tmpl_id.allow_partial_kit_delivery
+            )
+
+    def _prepare_phantom_move_values(self, bom_line, quantity):
+        """TODO: Not necessary in Odoo 13 or if this PR gets merged someday:
+        https://github.com/odoo/odoo/pull/67536"""
```

Comment:
How is this called for now?

Reply:
It's called in the kit explode method: https://github.com/odoo/odoo/blob/e46b6cefa5098447bd591b01148f0294428608ff/addons/mrp/models/stock_move.py#L169